.github/workflows/static_analysis.yml

```
@@ -34,9 +34,9 @@ jobs:
           path: |
             deps
             _build
-          key: ${{ runner.os }}-mix-otp-25-${{ hashFiles('**/mix.lock') }}
+          key: ${{ runner.os }}-${{ github.ref }}-mix-otp-25-${{ hashFiles('**/mix.lock') }}
           restore-keys: |
-            ${{ runner.os }}-mix-otp-25-
+            ${{ runner.os }}-${{ github.ref }}-mix-otp-25-
```

We kinda want to restore the cache only when `mix.lock` is the same here too, to prevent any issues later:
```suggestion
            ${{ runner.os }}-${{ github.ref }}-mix-otp-25-${{ hashFiles('**/mix.lock') }}
```